src/v/kafka/server/replicated_partition.cc

```diff
@@ -331,6 +331,9 @@ replicated_partition::timequery(storage::timequery_config cfg) {
         // The start override doesn't affect the result of the timequery.
         co_return res;
     }
+
+    // Local storage already handles this correctly. After we implement
+    // min_offset clamping for tiered storage, we can remove this code.
```

Comment:
nit: thanks for this comment, but it's not quite clear what handling this is referring to, and what code this referring to. Mind clarifying?

Reply:
I'm dropping this commit. Will update it in a different PR that is currently cooking.